### PR TITLE
GH-2330: Replace ExpiringMap with Cache API

### LINKED ISSUE #2330: Replace ExpiringMap with a Caffeine cache.
### Change

ExpiringMap is a custom written cache. ExpiringMap was taken from a separate project which isn't active.

Replace it with a Caffeine cache.

It is used in `GeometryLiteralIndex`, `GeometryTransformIndex` and `QueryRewriteIndex`.

### Are you interested in contributing a pull request for this task?

No

### PR DESCRIPTION
**Pull request by rvesse**
Replaces the usage of the `ExpiringMap` API with Jena's existing `Cache` abstraction API (backed by Caffeine `Cache`), this entails some related changes to the configuration portions of the GeoSPARQL module.

Main side effect of this change is that you can no longer actively change the max size/expiry interval of an index once it has been created.  This now requires replacing the existing index with a new one.

GitHub issue resolved #2330

----

 - [x] Tests are included.
 - [ ] Documentation change and updates are provided for the [Apache Jena website](https://github.com/apache/jena-site/)
 - [x] Commits have been squashed to remove intermediate development commit messages.
 - [x] Key commit messages start with the issue number (GH-xxxx, or if in JIRA, JENA-xxxx)

By submitting this pull request, I acknowledge that I am making a contribution to the Apache Software Foundation under the terms and conditions of the [Contributor's Agreement](https://www.apache.org/licenses/contributor-agreements.html).

### CODE REVIEW

**Comment by rvesse** [comment]
I don't know why GitHub says this branch is conflicted, tried rebasing and no difference 🤷

File: jena-geosparql/src/main/java/org/apache/jena/geosparql/implementation/index/CacheConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (afs) [inline jena-geosparql/src/main/java/org/apache/jena/geosparql/implementation/index/CacheConfiguration.java:1]: Wrong header.

It should be the 
"""
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.
""" 
header. 

See adjacent files for the text.

File: jena-geosparql/src/test/java/org/apache/jena/geosparql/implementation/index/CacheConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package org.apache.jena.geosparql.implementation.index;
```
Comment (afs) [inline jena-geosparql/src/test/java/org/apache/jena/geosparql/implementation/index/CacheConfigurationTest.java:1]: No header.

This is checked when running 

`mvn clean install -Pdev`

**Review by afs — CHANGES_REQUESTED** [review]

**Comment by afs** [comment]
@rvesse  Please could you change the commit message to `GH-2330: ...`. 2230 is something different. I've changed the pull request itself.

**Comment by rvesse** [comment]
> @rvesse Please could you change the commit message to `GH-2330: ...`. 2230 is pull request about something different. I've changed this pull request itself.

Done, and other comments addressed

**Review by afs — APPROVED** [review]

**Pull request merged**